drivers/sensor/max17262/max17262.c

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright 2020 Matija Tudan
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(max17262, CONFIG_SENSOR_LOG_LEVEL);
+
+#include "max17262.h"
+
+#define DT_DRV_COMPAT maxim_max17262
+
+/**
+ * @brief Read a register value
+ *
+ * Registers have an address and a 16-bit value
+ *
+ * @param priv Private data for the driver
+ * @param reg_addr Register address to read
+ * @param valp Place to put the value on success
+ * @return 0 if successful, or negative error code from I2C API
+ */
+static int max17262_reg_read(struct max17262_data *priv, int reg_addr,
+			     int16_t *valp)
+{
+	uint8_t i2c_data[2];
+	int rc;
+
+	rc = i2c_burst_read(priv->i2c, DT_INST_REG_ADDR(0), reg_addr,
+			    i2c_data, 2);
+	if (rc < 0) {
+		LOG_ERR("Unable to read register");
+		return rc;
+	}
+	*valp = (i2c_data[1] << 8) | i2c_data[0];
```

Comment:
Shouldn't `i2c_data[1]` be cast to `int16_t` first?

Reply:
Done.